python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -562,17 +564,24 @@ def partition_key(self) -> str:
     def asset_partition_key_for_output(self, output_name: str = "result") -> str:
         return self.partition_key
 
-    def asset_partitions_time_window_for_output(
-        self, output_name: str = "result"
-    ) -> Optional[TimeWindow]:
+    def asset_partitions_time_window_for_output(self, output_name: str = "result") -> TimeWindow:
+        return self.partition_time_window
+
+    @property
+    def partition_time_window(self) -> TimeWindow:
```

Comment:
This name makes it seem like this would work when there are no assets. Is that intended? Is that necessary for this change?

Reply:
ah... so in the execution context it does work for non-asset jobs. But in the direct invocation context it doesn't work for non-asset ops, since partitioned config is applied on the job level. I think we can remove this function from direct invocation and just ask users to use `asset_partitions_time_window_for_output` for the asset use case